packages/langium/src/workspace/workspace-manager.ts

```diff
@@ -116,7 +116,8 @@ export class DefaultWorkspaceManager implements WorkspaceManager {
                 if (entry.isDirectory) {
                     await this.traverseFolder(workspaceFolder, entry.uri, fileExtensions, collector);
                 } else if (entry.isFile) {
-                    const document = this.langiumDocuments.getOrCreateDocument(entry.uri);
+                    const content = await this.fileSystemProvider.readFile(entry.uri);
+                    const document = await this.langiumDocuments.createDocument(entry.uri, content, CancellationToken.None);
```

Comment:
In case we keep the current practice of parsing all documents upfront, we should change this method to first collect the URIs and then parse them in parallel. The current approach still parses them sequentially.

Reply:
But first we need to clarify the basic approach to LangiumDocuments (see discussion above).

Reply:
The current approach actualy parses them in parallel. See the `Promise.all(entry.map(...))` call.